### PR TITLE
hided nfc button from fast operations, created and hided default fiat…

### PR DESCRIPTION
**Pull request by Anschutz1927**
… button from settings, added donation functionality on portfolio, asset's settings, contacts, receive and wallet screens

### CODE REVIEW

File: app/src/main/java/io/multy/ui/adapters/ChainAdapter.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+        public Holder(View itemView) {
+            super(itemView);
+            ButterKnife.bind(this, itemView);
+        }
+    }
+
+    public enum ChainType {AVAILABLE, SOON}
+
+    public interface Listener {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/adapters/ChainAdapter.java]: Listener of what? Name it properly.

File: app/src/main/java/io/multy/ui/adapters/ChainAdapter.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        void onAvailableChainClick(String clickedChainName);
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/adapters/ChainAdapter.java]: start naming onlick methods with onClickWHAT or onItemClickWHAT- > onItemClickChain

File: app/src/main/java/io/multy/ui/fragments/BaseChooseFragment.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import io.multy.R;
+
+/**
+ * Created by anschutz1927@gmail.com on 05.03.18.
+ */
+
+public class BaseChooseFragment extends BaseFragment {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/BaseChooseFragment.java]: maybe BaseChooserFragment ?

File: app/src/main/java/io/multy/ui/fragments/BaseChooseFragment.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    @BindView(R.id.recycler_available)
+    RecyclerView recyclerAvailable;
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/BaseChooseFragment.java]: RecyclerView recyclerView;

File: app/src/main/java/io/multy/ui/fragments/asset/ChooseChainFragment.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import io.multy.R;
+import io.multy.ui.adapters.ChainAdapter;
+import io.multy.ui.fragments.BaseChooseFragment;
+import io.multy.ui.fragments.dialogs.DonateThisDialog;
+
+/**
+ * Created by anschutz1927@gmail.com on 03.03.18.
+ */
+
+public class ChooseChainFragment extends BaseChooseFragment implements ChainAdapter.Listener {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/asset/ChooseChainFragment.java]: name ChainChooserFragment. Read code convention again.

File: app/src/main/java/io/multy/ui/fragments/asset/ChooseCurrencyConvertFragment.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import io.multy.R;
+import io.multy.ui.adapters.ChainAdapter;
+import io.multy.ui.fragments.BaseChooseFragment;
+import io.multy.ui.fragments.dialogs.DonateThisDialog;
+
+/**
+ * Created by anschutz1927@gmail.com on 03.03.18.
+ */
+
+public class ChooseCurrencyConvertFragment extends BaseChooseFragment implements ChainAdapter.Listener {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/asset/ChooseCurrencyConvertFragment.java]: CurrencyChooserFragment

File: app/src/main/java/io/multy/ui/fragments/asset/ChooseCurrencyConvertFragment.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        if (getActivity() != null) {
+            DonateThisDialog.getInstance().show(getActivity().getSupportFragmentManager(), DonateThisDialog.TAG);
+        }
+    }
+
+    private void initialize(FragmentActivity activity) {
+        super.setTitle(R.string.currency_to_convert);
+        ChainAdapter chainAvailableAdapter = new ChainAdapter(ChainAdapter.ChainType.AVAILABLE, this);
+        ChainAdapter chainSoonAdapter = new ChainAdapter(ChainAdapter.ChainType.SOON, this);
+        getRecyclerAvailable().setLayoutManager(new LinearLayoutManager(activity));
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/asset/ChooseCurrencyConvertFragment.java]: What is recycler? is it View or not? where is the naming?

File: app/src/main/java/io/multy/ui/fragments/dialogs/DonateThisDialog.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import io.multy.R;
+import io.multy.ui.activities.DonationActivity;
+
+/**
+ * Created by anschutz1927@gmail.com on 05.03.18.
+ */
+
+public class DonateThisDialog extends DialogFragment {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/dialogs/DonateThisDialog.java]: what is "This"? DonateDialog use correct namings.

File: app/src/main/java/io/multy/ui/fragments/main/ChooseExchangeFragment.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import io.multy.R;
+import io.multy.ui.adapters.ExchangeAdapter;
+import io.multy.ui.fragments.BaseChooseFragment;
+import io.multy.ui.fragments.dialogs.DonateThisDialog;
+
+/**
+ * Created by anschutz1927@gmail.com on 05.03.18.
+ */
+
+public class ChooseExchangeFragment extends BaseChooseFragment implements ExchangeAdapter.Listener {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/main/ChooseExchangeFragment.java]: ExchangeChooserFragment

File: app/src/main/java/io/multy/ui/adapters/ChainAdapter.java [hunk 10]
```diff
@@ -138,8 +138,8 @@ public Holder(View itemView) {
 
     public enum ChainType {AVAILABLE, SOON}
 
-    public interface Listener {
-        void onAvailableChainClick(String clickedChainName);
-        void onSoonChainClick(String clickedChainName);
+    public interface OnClickListener {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/adapters/ChainAdapter.java]: Should be OnChainClickLister or on OnItemClickListener

File: app/src/main/java/io/multy/ui/adapters/CurrencyConvertAdapter.java [hunk 11]
```diff
@@ -100,20 +100,20 @@ public void setSoonChainsData(TypedArray chainsSoonImageIds, String[] chainsSoon
         notifyDataSetChanged();
     }
 
-    class HolderAvailable extends Holder {
+    class AvailableHolder extends Holder {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/adapters/CurrencyConvertAdapter.java]: Should be AvailableChainHolder

File: app/src/main/java/io/multy/ui/adapters/CurrencyConvertAdapter.java [hunk 12]
```diff
@@ -100,20 +100,20 @@ public void setSoonChainsData(TypedArray chainsSoonImageIds, String[] chainsSoon
...
             super(itemView);
             ButterKnife.bind(this, itemView);
         }
     }
 
-    class HolderSoon extends Holder {
+    class SoonHolder extends Holder {
 
-        HolderSoon(View itemView) {
+        SoonHolder(View itemView) {
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/adapters/CurrencyConvertAdapter.java]: Should be DisabledChainHolder or something similar

File: app/src/main/java/io/multy/ui/fragments/BaseChooserFragment.java [hunk 13]
```diff
@@ -24,12 +24,12 @@
  * Created by anschutz1927@gmail.com on 05.03.18.
  */
 
-public class BaseChooseFragment extends BaseFragment {
+public class BaseChooserFragment extends BaseFragment {
 
     @BindView(R.id.recycler_available)
-    RecyclerView recyclerAvailable;
+    RecyclerView availableRecyclerView;
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/BaseChooserFragment.java]: name of component should be first -> recyclerViewAvailableChains or similar

File: app/src/main/java/io/multy/ui/fragments/BaseChooserFragment.java [hunk 14]
```diff
@@ -24,12 +24,12 @@
...
     @BindView(R.id.recycler_soon)
-    RecyclerView recyclerSoon;
+    RecyclerView soonRecyclerView;
```
Comment (AndreyGrowl) [inline app/src/main/java/io/multy/ui/fragments/BaseChooserFragment.java]: the same as for available chains

File: app/src/main/java/io/multy/ui/fragments/BaseChooserFragment.java (hunk 13 above)
Comment (Anschutz1927) [reply app/src/main/java/io/multy/ui/fragments/BaseChooserFragment.java]: it's not only for chains fragment. it's base fragment.

File: app/src/main/java/io/multy/ui/fragments/BaseChooserFragment.java (hunk 14 above)
Comment (Anschutz1927) [reply app/src/main/java/io/multy/ui/fragments/BaseChooserFragment.java]: the same as for available chains

**Pull request merged**